js/damage.js

```diff
@@ -131,6 +131,12 @@ function getDamageResult(attacker, defender, move, field) {
     var typeEffect2 = defender.type2 ? getMoveEffectiveness(move, defender.type2, attacker.ability === "Scrappy" || field.isForesight, field.isGravity) : 1;
     var typeEffectiveness = typeEffect1 * typeEffect2;
     var PriorityDamageCheck = move.hasPriority && (defAbility === "Queenly Majesty" || defAbility === "Dazzling") && (["Mold Breaker", "Teravolt", "Turboblaze"].indexOf(attacker.ability) !== -1) || (move.givesHealth && attacker.ability === "triage");
+    var resistedKnockOffDamage = (defender.item === "" ||
+            (defender.name === "Giratina-Origin" && defender.item === "Griseous Orb") ||
+            (defender.name.indexOf("Arceus") !== -1 && defender.item.indexOf("Plate") !== -1) ||
+            (defender.name.indexOf("Genesect") !== -1 && defender.item.indexOf("Drive") !== -1) || 
+            (defender.ability.indexOf("RKS System") !== -1 && defender.item.indexOf("Memory") !== -1) || 
+            (defender.item.indexOf("Z") !== -1));
```

Comment:
@AustinXII Maybe I didn't make myself clear :/ I mean we can check `indexOf(" Z")`, in which there's a space in front of "Z", so that we can match all Z-Crystals and escape Zap Plate